File: Algorithms/Python3.x/69-Sqrt(x).py
# Runtime: 36 ms, faster than 95.79% of Python3 online submissions for Sqrt(x).
# Memory Usage: 13.4 MB, less than 7.56% of Python3 online submissions for Sqrt(x).
class Solution:
    def mySqrt(self, x: int) -> int:
        return int(x ** 0.5)


# Runtime: 52 ms, faster than 53.61% of Python3 online submissions for Sqrt(x).
# Memory Usage: 13.2 MB, less than 70.31% of Python3 online submissions for Sqrt(x).
class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, x
        while l <= r:
            mid =  l + (r - l) // 2
            if mid ** 2 <= x < (mid + 1) ** 2:
                return mid
            elif x < mid ** 2:
                r = mid
            else:
                l = mid + 1

File: Algorithms/C++/69-Sqrt(x).cpp
// Runtime: 4 ms, faster than 75.92% of C++ online submissions for Sqrt(x).
// Memory Usage: 8.2 MB, less than 98.25% of C++ online submissions for Sqrt(x).
class Solution {
public:
    int mySqrt(int x) {
        return sqrt(x);
    }
};


// Runtime: 24 ms, faster than 9.75% of C++ online submissions for Sqrt(x).
// Memory Usage: 8.2 MB, less than 100.00% of C++ online submissions for Sqrt(x).
class Solution {
public:
    int mySqrt(int x) {
        long int i = 0;
        while (i * i <= x)
            ++i;
        return i - 1;
    }
};
